### PR TITLE
Add TLS support to CoAP adapter and all readers

### PR DESCRIPTION
**Pull request by anovakovic01**
### What does this do?
Adds TLS client support for CoAP adapter and all readers.

### Have you included tests for your changes?
No.

### Did you document any new/modified functionality?
Yes, I updated `README.md` files.

### CODE REVIEW

**Review by nwest1 — DISMISSED** [review]
I do like the use of a flag in the TLS configuration!

**Comment by drasko** [comment]
@nwest1 usually the goal here is for @anovakovic01 to see this picture: 
image

You have been too soft on him, there must be something to request changes ;)

File: cmd/cassandra-reader/main.go [hunk 1]
```diff
@@ -103,13 +118,27 @@ func connectToCassandra(cluster, keyspace string, logger logger.Logger) *gocql.S
 	return session
 }
 
-func connectToThings(url string, logger logger.Logger) *grpc.ClientConn {
-	conn, err := grpc.Dial(url, grpc.WithInsecure())
+func connectToThings(cfg config, logger logger.Logger) *grpc.ClientConn {
+	var opts []grpc.DialOption
+	if cfg.clientTLS {
+		if cfg.caCerts != "" {
+			tpc, err := credentials.NewClientTLSFromFile(cfg.caCerts, "")
```
Comment (drasko) [inline cmd/cassandra-reader/main.go:125]: What's `tpc`? If I did not get this on a first sight, would that mean that naming is badly chosen?

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by chombium — COMMENTED** [review]
The pr looks good. 
I have one small comment. In each strconv.ParseBool error check if the envClientTLS var contains a string which can not be converted to bool (user typo when setting the envvar's value, for example), the TLS encryption is silently turned off.
Later in the code, the user will get a message that the encryption is turned off, but it might be confusing why: both TLS envvars are set, but no TLS is activated.

IMO it would be better if a message is logged sayng that the envvar's value is wrong. The other thing to think about is if we want to throw an error and exit, because of the misconfigured envvar, or we want to do our best and start the component in a way which is not wanted by the user (that's what's currently being done).

What do you think about that?

File: cmd/cassandra-reader/main.go (hunk 1 above)
Comment (anovakovic01) [reply cmd/cassandra-reader/main.go:125]: We have this name in every service, and it's short for `TransportCredentials`. Also, this is local var that is used in only one place, so this name is not so important (it is in go spirit to use short names for local vars).

**Comment by anovakovic01** [comment]
@chombium Thanks for the feedback, I was not sure about this myself. Now that I think about this, I think that we should exit with logged error, if TLS env var has invalid value.

**Comment by chombium** [comment]
@anovakovic01 thanks for the quick response. That looks much better

**Review by drasko — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**